### PR TITLE
Fix jank and large-jumping frame by controlling rasterizer ending time (V2)

### PR DESCRIPTION
**Pull request by fzyzcjy**
Previous: https://github.com/flutter/engine/pull/36837

### CODE REVIEW

File: shell/common/rasterizer.cc [hunk 1]
```diff
@@ -804,6 +812,31 @@ std::optional<size_t> Rasterizer::GetResourceCacheMaxBytes() const {
...
+    // Use busy wait instead of thread sleep, because otherwise we cannot
+    // guarantee we are scheduled at the desired time by OS.
+    // It may be seemingly wasting to busy wait, but indeed it is not. The
+    // sleep only happens under the scenario discussed in #36837, i.e. current
+    // rasterization is significantly shorter than other rasterizations.
+    // Then, we are just using the CPU cycles that should have been used in
+    // normal cases.
+    while (fml::TimePoint::Now() < wakeup_time.value()) {
+      // nothing here
+    }
```
Comment (dnfield) [inline shell/common/rasterizer.cc:836]: I am highly skeptical that this will ever be an acceptable approach.

**Review by dnfield — CHANGES_REQUESTED** [review]
I don't think this is currently in a good state to review - as discussed, we need some benchmarks that would consistently show where this will help and have good reasons to believe it's not making other problems worse.

I'm going to close this to remove it from review queues for now.

**Comment by fzyzcjy** [comment]
All right, I will try to squeeze some time to make a benchmark later :)

**Comment by fzyzcjy** [comment]
As discussed in Discord with @dnfield previously, I will firstly focus on the PRs that are critical to the main part of flutter_smooth, and deal with these PRs (which improves performance roughly several FPS per PR) later.

**Pull request closed without merging**